testutil/common/tester.go

```diff
@@ -57,36 +62,89 @@ func NewTester(t *testing.T) *Tester {
 	return ts
 }
 
-func (ts *Tester) AddAccount(name string, balance int64) *Tester {
-	ts.accounts[name] = CreateNewAccount(ts.GoCtx, *ts.Keepers, balance)
-	return ts
+func (ts *Tester) SetChainID(chainID string) {
+	blockHeader := ts.Ctx.BlockHeader()
+	blockHeader.ChainID = chainID
+	ts.Ctx = ts.Ctx.WithBlockHeader(blockHeader)
+	ts.GoCtx = sdk.WrapSDKContext(ts.Ctx)
 }
 
 func (ts *Tester) SetupAccounts(numSub, numAdm, numDev int) *Tester {
-	for i := 0; i < numSub; i++ {
-		name := "sub" + strconv.Itoa(i+1)
-		ts.AddAccount(name, 20000)
+	for i := 1; i <= numSub; i++ {
+		ts.AddAccount("sub", i, 20000)
 	}
-	for i := 0; i < numAdm; i++ {
-		name := "adm" + strconv.Itoa(i+1)
-		ts.AddAccount(name, 10000)
+	for i := 1; i <= numAdm; i++ {
+		ts.AddAccount("adm", i, 10000)
 	}
-	for i := 0; i < numDev; i++ {
-		name := "dev" + strconv.Itoa(i+1)
-		ts.AddAccount(name, 10000)
+	for i := 1; i <= numDev; i++ {
+		ts.AddAccount("dev", i, 10000)
 	}
 
 	return ts
 }
 
+func (ts *Tester) AddAccount(kind string, idx int, balance int64) (Account, string) {
+	name := kind + strconv.Itoa(idx)
+	ts.accounts[name] = CreateNewAccount(ts.GoCtx, *ts.Keepers, balance)
+	return ts.Account(name)
+}
+
+func (ts *Tester) GetAccount(kind string, idx int) (Account, string) {
+	name := kind + strconv.Itoa(idx)
+	return ts.Account(name)
+}
+
 func (ts *Tester) Account(name string) (Account, string) {
 	account, ok := ts.accounts[name]
 	if !ok {
-		panic("tester: unknown account: '" + name + "'")
+		panic("tester: unknown account name: '" + name + "'")
 	}
 	return account, account.Addr.String()
 }
 
+func (ts *Tester) Accounts(name string) []Account {
+	var names []string
+	for k := range ts.accounts {
+		if strings.HasPrefix(k, name) {
+			names = append(names, k)
+		}
+	}
+	sort.Strings(names)
+	var accounts []Account
+	for _, k := range names {
+		accounts = append(accounts, ts.accounts[k])
+	}
+	return accounts
+}
+
+func (ts *Tester) StakeAccount(acct Account, spec spectypes.Spec, amount int64, geoloc uint64, moniker string) error {
```

Comment:
is this used?
suppose to be deprecated

Reply:
this stakes a provider, given its common.Account.
similar to StakeProvider(), except the latter takes the string address, i.e. acc.Addr.String()
it gets some clean-up in the PR of part 5; but I can take a step further and replace with StakeProvider where used.

Reply:
fixed 4d40a32c